src/index.ts

```diff
@@ -0,0 +1,212 @@
+  CLOUDFLARE_API_TOKEN: getInput("apiToken"),
+  CLOUDFLARE_ACCOUNT_ID: getInput("accountId"),
+  ENVIRONMENT: getInput("environment"),
+  VARS: getMultilineInput("vars"),
+  COMMANDS: getMultilineInput("commands"),
+} as const;
+
+async function main() {
+  installWrangler();
+  authenticationSetup();
+  await execCommands(getMultilineInput("preCommands"));
+  await uploadSecrets();
+  await wranglerCommands();
+  await execCommands(getMultilineInput("postCommands"));
+}
+
+function checkWorkingDirectory(workingDirectory = "") {
+  try {
+    const normalizedPath = path.normalize(workingDirectory);
+    if (existsSync(normalizedPath)) {
+      return normalizedPath;
+    } else {
+      setFailed(`🚨 Directory ${workingDirectory} does not exist.`);
+    }
+  } catch (error) {
+    setFailed(
+      `🚨 While checking/creating directory ${workingDirectory} received ${error}`
+    );
+  }
+}
+
+function installWrangler() {
+  if (config["WRANGLER_VERSION"].startsWith("1")) {
+    setFailed(`🚨 Wrangler v1 is no longer supported by this action. Please use major versions 2 or 3`);
+  }
+  startGroup("📥 Installing Wrangler");
+  const command = `npx install wrangler@${config["WRANGLER_VERSION"]}`;
+  info(`Running Command: ${command}`);
+  spawnSync(command, { cwd: config["workingDirectory"], env: process.env });
+  endGroup();
+}
+
+function authenticationSetup() {
+  startGroup("🔐 Authenticating with Cloudflare");
+  try {
+    process.env.CLOUDFLARE_API_TOKEN = config["CLOUDFLARE_ACCOUNT_ID"];
+    process.env.CLOUDFLARE_ACCOUNT_ID = config["CLOUDFLARE_API_TOKEN"];
+    info(`Authentication process initiated with - API Token`);
+  } catch (error) {
+    setFailed(
+      `Authentication details were not found. Please input an 'apiToken' to the action.`
+    );
+  }
+  endGroup();
+}
+
+async function execCommands(commands: string[]) {
+  if (!commands.length) {
+    return;
+  }
+
+  startGroup("🚀 Pre/Post Commands Group");
+
+  const arrPromises = commands.map(async (command) => {
+    const npxCMD = command.startsWith("wrangler") ? `npx ${command}` : command;
+
+    info(`🚀 Executing command: ${npxCMD}`);
+
+    await exec(npxCMD, {
+      cwd: config["workingDirectory"],
+      env: process.env,
+    });
+  });
+
+  await Promise.all(arrPromises);
+  endGroup();
+}
+
+async function uploadSecrets() {
+  const secrets = config["secrets"];
+  if (!secrets.length) {
+    return;
+  }
+
+  startGroup("🔑 Uploading Secrets");
+  const npxCMD = process.env.RUNNER_OS === "Windows" ? "npx.cmd exec" : "npx";
+  const environment = config["ENVIRONMENT"];
+
+  const getSecret = (secret: string) =>
+    process.env[secret] ?? setFailed("🚨 Secret not found in env");
+
+  const environmentSuffix = !environment.length ? "" : ` --env ${environment}`;
+
+  await Promise.all(
+    secrets.map(async (secret) =>
+      await exec(
+        `${npxCMD} wrangler secret put ${getSecret(
+          secret
+        )}${environmentSuffix}`,
```

Comment:
I could be wrong, but this looks slightly off. It seems like it's running `npx wrangler secret put SECRET_VALUE` when it should be running `npx wrangler secret put SECRET_NAME` and then piping in the value? (Or using `secret:bulk`)

Reply:
I wanted to use the secrets:bulk but it only accepts JSON files, for some reason I thought I had made it do both.

> when it should be running npx wrangler secret put SECRET_NAME and then piping in the value
...ah man, yeah, thanks for the catch.